### PR TITLE
Debug updates

### PR DESCRIPTION
**Pull request by de-vri-es**
This continues from #1099. Credits to @jnmaloof for the original work. It appears they are limited in available time, so I figured I'd address the remaining comments to get it merged.

@jnmaloof: Would this satisfy your use case? I added a `--flatten-output` option to allow flattening like your original PR.

Note: I haven't gotten round to test the path generation yet. I'll try to do that soon.

### CODE REVIEW

**Comment by jnmaloof** [comment]
Yes.  Sorry that I dropped this.  Fall quarter started and I am teaching, so this got back burnered.  Thanks for picking it up.

File: keras_retinanet/bin/debug.py [hunk 1]
```diff
@@ -172,6 +181,9 @@ def csv_list(string):
     parser.add_argument('--image-min-side', help='Rescale the image so the smallest side is min_side.', type=int, default=800)
     parser.add_argument('--image-max-side', help='Rescale the image if the largest side is larger than max_side.', type=int, default=1333)
     parser.add_argument('--config', help='Path to a configuration parameters .ini file.')
+    parser.add_argument('--no-gui', help='Do not open a GUI window. Save images to an output directory instead.', action='store_true')
+    parser.add_argument('--output-dir', help='if --no-gui seleceted, specify directory base for saving images.', default='.')
```
[the remark this answers is not in the document]
Comment (de-vri-es) [reply keras_retinanet/bin/debug.py]: Good point, fixed.

File: keras_retinanet/bin/debug.py [hunk 2]
```diff
@@ -172,6 +181,9 @@ def csv_list(string):
...
+    parser.add_argument('--flatten-output', help='if --no-gui seleceted, flatten the folder structure of saved output images into a single folder.', action='store_true')
```
[the remark this answers is not in the document]
Comment (de-vri-es) [reply keras_retinanet/bin/debug.py]: Also fixed.

File: keras_retinanet/bin/debug.py [hunk 3]
```diff
@@ -241,6 +264,18 @@ def run(generator, args, anchor_params):
     return True
 
 
+def make_output_path(output_dir, image_path, flatten = False):
```
[the remark this answers is not in the document]
Comment (de-vri-es) [reply keras_retinanet/bin/debug.py]: Done :)

**Comment by de-vri-es** [comment]
There was one bug remaining: I used `path` somewhere instead of `os.path`. Fixed now. I think it's ready to merge.

@hgaiser: do you agree?

**Comment by hgaiser** [comment]
LGTM :+1:

**Pull request merged**